session/session.go

```diff
@@ -69,6 +85,7 @@ type Session struct {
 	lastTime          int64                  // last heartbeat time
 	entity            NetworkEntity          // low-level network entity
 	data              map[string]interface{} // session data store
+	handshakeData     *HandshakeData         // handshake data received by the client
```

Comment:
Why didn't you put the handshake data on session's data?

Reply:
That way the user would have to keep in mind of the handshake data when changing the session data. For example, clearing the session data would also clear the handshake data, this not being very obvious.